crates/oct-cli/src/main.rs

```diff
@@ -39,37 +44,67 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
 
     let cli = Cli::parse();
 
+    let state_file_path = "./state.json";
+
     match &cli.command {
         Commands::Deploy(args) => {
             // Create EC2 instance
             let mut instance = aws::Ec2Instance::new(
                 "us-west-2".to_string(),
                 "ami-04dd23e62ed049936".to_string(),
-                aws::aws_sdk_ec2::types::InstanceType::T2Micro,
+                aws::aws_sdk_ec2::types::InstanceType::T2Micro.to_string(),
                 "oct-cli".to_string(),
             )
             .await;
 
             instance.create().await?;
 
+            // Save to state file
+            let instance_state = Ec2InstanceState {
+                id: instance.id.clone().unwrap_or_default(),
+                arn: instance.arn.clone().unwrap_or_default(),
+                public_ip: instance.public_ip.clone().unwrap_or_default(),
+                public_dns: instance.public_dns.clone().unwrap_or_default(),
+                region: instance.region.clone(),
+                ami: instance.ami.clone(),
+                instance_type: instance.instance_type.clone(),
+                name: instance.name.clone(),
+                instance_profile: instance.instance_profile.as_ref().map(|ip| {
+                    InstanceProfileState {
+                        name: ip.name.clone(),
+                        region: ip.region.clone(),
+                        instance_roles: ip
+                            .instance_roles
+                            .iter()
+                            .map(|ir| InstanceRoleState {
+                                name: ir.name.clone(),
+                                region: ir.region.clone(),
+                                assume_role_policy: ir.assume_role_policy.clone(),
+                                policy_arns: ir.policy_arns.clone(),
+                            })
+                            .collect(),
+                    }
+                }),
+            };
```

Comment:
Let's create functions `to_state` in Ec2Instance, InstanceProfile and InstanceRole that return Ec2InstanceState, InstanceProfileState and InstanceRoleState respectively